src/go/configgenerator/listener_generator.go

```diff
@@ -169,6 +169,11 @@ func makeListener(serviceInfo *sc.ServiceInfo) (*v2pb.Listener, error) {
 	}
 
 	httpConMgr := &hcmpb.HttpConnectionManager{
+		UpgradeConfigs: []*hcmpb.HttpConnectionManager_UpgradeConfig{
+			{
+				UpgradeType: "websocket",
+			},
+		},
```

Comment:
Shall we iterate all the backend and if they are all `isHttp2`(I believe that more common that all backends are all http1 or all http2), skip configging websocket?

Reply:
Envoy will handle websockets on HTTP2 as well, so I don't think we need to skip the config.

Reply:
Agreed that we don't need to skip http2 backend.  According [this](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/http/websocket#arch-overview-websocket)  upgrade is disabled by default for http2.